### ISSUE TITLE
In POSIX sh, trap flags are undefined

### ISSUE THREAD

**Issue by contivero** [koalaman/shellcheck#1461]
#### For new checks and feature suggestions
- [X] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
trap -l
```

#### Here's what shellcheck currently says:
Nothing

#### Here's what I wanted or expected to see:
"In POSIX sh, trap flags are undefined"

There are no flags for the `trap` builtin defined under POSIX sh ([see spec](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#trap)), and the same is true for dash, but bash defines the flags [-lp].
